src/core/src/bootstrap/EnvLayer.py

```diff
@@ -242,6 +242,11 @@ def cpu_arch():  # architecture
         @staticmethod
         def vm_name():     # machine name
             return platform.node()
+
+        @staticmethod
+        def version():
+            return str() if (EnvLayer.get_python_major_version() == 2) else distro.os_release_attr('version')
```

Comment:
We carry our own copy of distro - why is this necessary?

Also, this needs to be more reliable --
return (distro.os_release_attr('version') or
        distro.os_release_attr('version_id') or
        distro.version() or
        Constants.Version.Unknown)   // use correct constant equivalent - this is just illustrative

Aside from that, a general comment at envlayer - the code must be deterministic for a caller. Hidden 2 vs 3 behavior needs to always be thought through more.

Reply:
This was added to the PR in error, I was trying something specific to Azure Linux back then

Reply:
Removed